src/OVAL/probes/unix/sysctl.c

```diff
@@ -149,9 +165,21 @@ int probe_main(probe_ctx *ctx, void *probe_arg)
                         l = fread(sysval, 1, sizeof sysval - 1, fp);
 
                         if (ferror(fp)) {
-                                dE("An error ocured when reading from \"%s\" (fp=%p): l=%ld, %u, %s\n",
-                                    mibpath, fp, l, errno, strerror(errno));
-                                goto fail_item;
+				/* Linux 4.1.0 introduced a per-NIC IPv6 stable_secret file.
+				 * The stable_secret file cannot be read until it is set,
+				 * so we skip it.
+				 */
+				if (strncmp(ofts_ent->path, ipv6_conf_path, ipv6_conf_path_len) == 0 &&
```

Comment:
Wouldn't it make sense to check for this path before we attempt to read it? If we are sure it will cause an error we want to avoid even attempting to read it.